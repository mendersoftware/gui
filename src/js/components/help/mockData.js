// Copyright 2020 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {
    hasDeviceConfig: true,
    hasMonitor: true
  },
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '2003-01-01'
  }
};
