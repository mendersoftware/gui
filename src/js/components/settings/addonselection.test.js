import React from 'react';

import { undefineds } from '../../../../tests/mockData';
import { render } from '../../../../tests/setupTests';
import AddOnSelection from './addonselection';

describe('AddOnSelection component', () => {
  it('renders correctly', () => {
    const { baseElement } = render(
      <AddOnSelection addOns={[{ name: 'configure', enabled: true }]} features={{ hasDeviceConfig: true, hasDeviceConnect: true }} />
    );
    const view = baseElement.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
