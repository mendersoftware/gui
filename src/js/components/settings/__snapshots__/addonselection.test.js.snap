// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddOnSelection component renders correctly 1`] = `
.emotion-0 {
  border-color: #f7f7f7;
}

.emotion-0.active,
.emotion-0:hover {
  border-color: #d4e9e7;
  box-shadow: 0 1px 6px rgba(0, 0, 0, 0.15);
}

.emotion-0.active {
  background-color: #f7f7f7;
}

.emotion-0.addon {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 10px;
  column-gap: 10px;
  display: grid;
  grid-template-columns: max-content minmax(150px, min-content) 1fr max-content;
  height: initial;
  margin-top: 10px;
  min-width: 550px;
  width: initial;
}

.emotion-0.addon.upgrade {
  grid-template-columns: max-content minmax(120px, min-content) minmax(120px, min-content) 1fr max-content;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  padding: 9px;
  border-radius: 50%;
  color: rgba(0, 0, 0, 0.54);
  color: rgba(10, 10, 11, 0.78);
}

.emotion-1::-moz-focus-inner {
  border-style: none;
}

.emotion-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-1:hover {
  background-color: rgba(51, 122, 135, 0.04);
}

@media (hover: none) {
  .emotion-1:hover {
    background-color: transparent;
  }
}

.emotion-1.Mui-checked,
.emotion-1.MuiCheckbox-indeterminate {
  color: #337a87;
}

.emotion-1.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-2 {
  cursor: inherit;
  position: absolute;
  opacity: 0;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.emotion-3 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.3929rem;
}

.emotion-3 iconButton {
  margin-right: 8px;
}

.emotion-4 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-5 {
  font-size: 1rem;
}

.emotion-6 {
  color: rgba(0, 0, 0, 0.38);
  margin: 15px 0;
}

<div>
  <h3
    class="margin-top-large"
  >
    Get more features with add-ons
  </h3>
  <div
    class="flexbox column"
  >
    <p>
      Extend Mender features with our add-ons. Select one or more from the list and submit to request add-ons to be added to your plan. We'll adjust your subscription and confirm it with you.
    </p>
    <div
      class="planPanel emotion-0 addon upgrade  "
    >
      <span
        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary emotion-1"
      >
        <input
          class="PrivateSwitchBase-input emotion-2"
          data-indeterminate="false"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root emotion-4"
        />
      </span>
      <div
        class="bold"
      >
        Configure
      </div>
      <div
        class="flexbox column"
      >
        <div
          class="link-color bold emotion-5"
        >
          $11/month
        </div>
        <div>
          for first 50 devices
        </div>
      </div>
      <span
        class="emotion-6 "
      >
        Expand your plan with device configuration features
      </span>
      <a
        class="margin-left-small"
        href="https://mender.io/plans/features"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn more
      </a>
    </div>
    <div
      class="planPanel emotion-0 addon upgrade  "
    >
      <span
        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary emotion-1"
      >
        <input
          class="PrivateSwitchBase-input emotion-2"
          data-indeterminate="false"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root emotion-4"
        />
      </span>
      <div
        class="bold"
      >
        Troubleshoot
      </div>
      <div
        class="flexbox column"
      >
        <div
          class="link-color bold emotion-5"
        >
          $27/month
        </div>
        <div>
          for first 50 devices
        </div>
      </div>
      <span
        class="emotion-6 "
      >
        Expand your plan with device troubleshooting features
      </span>
      <a
        class="margin-left-small"
        href="https://mender.io/plans/features"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn more
      </a>
    </div>
  </div>
</div>
`;
